doc/source/development/contributing_code_style.rst

```diff
@@ -0,0 +1,199 @@
+.. _Not_sure_what_to_put_here:
+
+{{ header }}
+
+=======================
+pandas code style guide
+=======================
+
+.. contents:: Table of contents:
+   :local:
+
+Patterns
+========
+
+foo.__class__
+-------------
+
+*pandas* uses 'type(foo)' instead 'foo.__class__' as it is making the code more
+readable.
+
+For example:
+
+**Good:**
+
+.. code-block:: python
+
+    cls = type(self)
+
+**Good:**
+
+.. code-block:: python
+
+    name = type(self).__name__
+
+**Bad:**
+
+.. code-block:: python
+
+    cls = self.__class__
+
+**Bad:**
+
+.. code-block:: python
+
+    name = self.__class__.__name__
+
+String formatting
+=================
+
+Concatenated strings
+--------------------
+
+f-strings
+~~~~~~~~~
+
+*pandas* uses f-strings formatting instead of '%' and '.format()' string formatters.
+
+The convention of using f-strings on a string that is concatenated over serveral lines,
+is to prefix only the lines containing the value needs to be interpeted.
+
+For example:
+
+**Good:**
+
+.. code-block:: python
+
+    my_warning_message = (
+        f"Warning, {old_function_name} is deprecated, "
+        "please use the new and way better "
+        f"{new_function_name}"
+    )
+
+**Bad:**
+
+.. code-block:: python
+
+    my_warning_message = (
+        f"Warning, {old_function_name} is deprecated, "
+        f"please use the new and way better "
+        f"{new_function_name}"
+    )
+
+White spaces
+~~~~~~~~~~~~
+
+Putting the white space only at the end of the previous line, so
+there is no whitespace at the beggining of the concatenated string.
+
+For example:
+
+**Good:**
+
+.. code-block:: python
+
+    example_string = (
+        "Some long concatenated string, "
+        "with good placement of the "
+        "whitespaces"
+    )
+
+**Bad:**
+
+.. code-block:: python
+
+    example_string = (
+        "Some long concatenated string,"
+        " with bad placement of the"
+        " whitespaces"
+    )
+
+Representation function (aka 'repr()')
+--------------------------------------
+
+*pandas* uses 'repr()' instead of '%r' and '!r'.
+
+The use of 'repr()' will only happend when the value is not an obvious string.
```

Comment:
I don't think this is true. In some cases it makes more sense to represent the object as a string, and in some as its representation. See:
```python
>>> f'today is {my_date}'
'today is 2020-01-01'
>>> f'the object {my_date!r} is not a string'
'the object datetime.date(2020, 1, 1) is not a string'
```
Better avoid saying when to use the representation, since this is more common sense, and focus on the formatting.

@jbrockmendel do we really prefer `{repr(foo)}` to `{foo!r}`. I think I've seen some discussion about this, but not sure this is really an standard we follow, and why the former should be preferred.